src/composer.rs

```diff
@@ -749,6 +749,23 @@ impl Composer {
         );
     }
 
+    /// Subtracts a curve point from another by consuming 3 gates.
+    pub fn component_sub_point(
+        &mut self,
+        a: WitnessPoint,
+        b: WitnessPoint,
+    ) -> WitnessPoint {
+        // We negate the 'x' coordinate of the point 'b', so that neg_b = (-b.x,
+        // b.y)
+        let constraint = Constraint::new().left(-BlsScalar::one()).a(*b.x());
+        let neg_b_x = self.gate_mul(constraint);
+
+        let neg_b = WitnessPoint::new(neg_b_x, *b.y());
```

Comment:
I would even go as far as to add a separate `component_neg_point` as well so that we can use this in more places.